libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -1362,106 +1365,571 @@ def _extract_images_from_page(self, page: pypdfium2._helpers.page.PdfPage) -> st
+                                    "page": page.page_number - 1,
+                                }
+                            )
+                        ),
+                    )
+                else:
+                    contents.append(all_text)
+                # "tables_as_html": [self._convert_table_to_html(table)
+                #                    for
+                #                    table in tables_content],
+                # "images": images_content,
+                # tables_as_html.extend([self._convert_table(table)
+                #                        for
+                #                        table in tables_content])
+            if self.mode == "single":
+                yield Document(
+                    page_content=self.pages_delimiter.join(contents),
+                    metadata=_validate_metadata(
+                        _PDFPlumberParserMetadata(doc_metadata)
                     ),
                 )
-                for page in doc.pages
-            ]
 
     def _process_page_content(self, page: pdfplumber.page.Page) -> str:
-        """Process the page content based on dedupe."""
+        """Process the page content based on dedupe.
+
+        Args:
+            page: The PDF page to process.
+
+        Returns:
+            The extracted text from the page.
+        """
         if self.dedupe:
             return page.dedupe_chars().extract_text(**self.text_kwargs)
         return page.extract_text(**self.text_kwargs)
 
-    def _extract_images_from_page(self, page: pdfplumber.page.Page) -> str:
-        """Extract images from page and get the text with RapidOCR."""
+    def _split_page_content(
+        self,
+        page: pdfplumber.page.Page,
+        tables_bbox: list[tuple[float, float, float, float]],
+        tables_content: list[list[list[Any]]],
+        images_bbox: list[tuple[float, float, float, float]],
+        images_content: list[np.ndarray],
+        **kwargs: Any,
+    ) -> Iterator[Union[str, list[list[str]], np.ndarray]]:
+        """Split the page content into text, tables, and images.
+
+        Args:
+            page: The PDF page to process.
+            tables_bbox: Bounding boxes of tables on the page.
+            tables_content: Content of tables on the page.
+            images_bbox: Bounding boxes of images on the page.
+            images_content: Content of images on the page.
+            **kwargs: Additional keyword arguments.
+
+        Yields:
+            An iterator over the split content (text, tables, images).
+        """
+        from pdfplumber.utils import (
+            geometry,
+            text,
+        )
+
+        # Iterate over words. If a word is in a table,
+        # yield the accumulated text, and the table
+        # A the word is in a previously see table, ignore it
+        # Finish with the accumulated text
+        kwargs.update(
+            {
+                "keep_blank_chars": True,
+                # "use_text_flow": True,
```

Comment:
There's a lot of commented out code in this implementation -- would you be able to remove it?